safenode/src/network/event.rs

```diff
@@ -199,11 +199,24 @@ impl SwarmDriver {
             SwarmEvent::ConnectionEstablished {
                 peer_id, endpoint, ..
             } => {
+                debug!(
+                    "Connection established with {peer_id:?}. We are the dialer {:?}",
+                    endpoint.is_dialer()
+                );
                 if endpoint.is_dialer() {
                     info!("Connected with {peer_id:?}");
                     if let Some(sender) = self.pending_dial.remove(&peer_id) {
                         let _ = sender.send(Ok(()));
                     }
+                } else {
+                    // If the connection was opened by someone else / relayed to us,
+                    // then add the peer to the routing table.
+                    let _routing_update = self
+                        .swarm
+                        .behaviour_mut()
+                        .kademlia
+                        .add_address(&peer_id, endpoint.get_remote_address().clone());
```

Comment:
In theory, if this is a relayed connection, then we shall able to dial back using the `endpoint.get_remote_address()` ?

So, here we shall try a dial back, and if the dial succeeded, we then can add peer into RT.
instead of adding it to RT directly ?